cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectAssemblyOperator.java

```diff
@@ -84,13 +93,89 @@ public KafkaConnectAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pf
         annotations.put(ANNO_STRIMZI_IO_LOGGING, logAndMetricsConfigMap.getData().get(connect.ANCILLARY_CM_KEY_LOG_CONFIG));
 
         log.debug("{}: Updating Kafka Connect cluster", reconciliation, name, namespace);
-        return  connectServiceAccount(namespace, connect)
+        Future<Void> chainFuture = Future.future();
+        connectServiceAccount(namespace, connect)
                 .compose(i -> deploymentOperations.scaleDown(namespace, connect.getName(), connect.getReplicas()))
                 .compose(scale -> serviceOperations.reconcile(namespace, connect.getServiceName(), connect.generateService()))
                 .compose(i -> configMapOperations.reconcile(namespace, connect.getAncillaryConfigName(), logAndMetricsConfigMap))
                 .compose(i -> podDisruptionBudgetOperator.reconcile(namespace, connect.getName(), connect.generatePodDisruptionBudget()))
                 .compose(i -> deploymentOperations.reconcile(namespace, connect.getName(), connect.generateDeployment(annotations, pfa.isOpenshift(), imagePullPolicy, imagePullSecrets)))
-                .compose(i -> deploymentOperations.scaleUp(namespace, connect.getName(), connect.getReplicas()).map((Void) null));
+                .compose(i -> deploymentOperations.scaleUp(namespace, connect.getName(), connect.getReplicas()))
+                .compose(i -> deploymentOperations.readiness(namespace, connect.getName(), 1_000, this.operationTimeoutMs))
+                .compose(i -> chainFuture.complete(), chainFuture)
+                .setHandler(reconciliationResult -> {
+                    StatusUtils.setStatusConditionAndObservedGeneration(kafkaConnect, kafkaConnectStatus, reconciliationResult);
+                    kafkaConnectStatus.setRestApiAddress(connect.getServiceName() + "." + namespace + ".svc:" + KafkaConnectCluster.REST_API_PORT);
+
+                    updateStatus(kafkaConnect, reconciliation, kafkaConnectStatus).setHandler(statusResult -> {
+                        // If both features succeeded, createOrUpdate succeeded as well
+                        // If one or both of them failed, we prefer the reconciliation failure as the main error
+                        if (reconciliationResult.succeeded() && statusResult.succeeded()) {
+                            createOrUpdateFuture.complete();
+                        } else if (reconciliationResult.failed()) {
+                            createOrUpdateFuture.fail(reconciliationResult.cause());
+                        } else {
+                            createOrUpdateFuture.fail(statusResult.cause());
+                        }
```

Comment:
Am I wrong or these few lines of code are the same for the other assembly operators handling the reconcile and status result? Could we factor out a common method in the base class?

Reply:
You are not wrong. We agreed with @scholzj to refactor common code after all status resources will be merged.